### ISSUE TITLE
SC1058 false positive when `do` keyword is in line continuation

### ISSUE THREAD

**Issue by Lin-Buo-Ren** [koalaman/shellcheck#926]
- Rule Id: SC1058(SC1073)
- My shellcheck version (`shellcheck --version` or "online"): online & 0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bash
for required_command in\
	mount\
	grep\
	dd\
	do
	if ! command -v "${required_command}" >/dev/null; then
		echo "Error: Command ${required_command} not found."
	fi
done
```

#### Here's what shellcheck currently says:
```
Line 2:
for required_command in\
^-- SC1073: Couldn't parse this for loop.
                     ^-- SC1058: Expected 'do'.
                       ^-- SC1072: Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:
```
No issues detected!
```

**Comment by koalaman** [comment]
This is indeed not very helpful. This variation is a bit more correctly diagnosed, though it doesn't mention that the escaped linefeed is insufficient: 

```
#!/usr/bin/env bash
for required_command in mount\
  grep\
  dd\
  do
  ^-- SC1063: You need a line feed or semicolon before the 'do'.
  if ! command -v "${required_command}" >/dev/null; then
    echo "Error: Command not found."
  fi
done
```

It's not valid in bash so there should be some error.

**Comment by kurahaupo** [comment]
On Mon, 5 Jun 2017, koalaman wrote:
> It's not valid in bash so there should be some error.

Maybe the check-list should include another question:

"I've tried running this sample code in Bash version ____.

 [_] The script behaved as expected (expected exit code, expected output, and NO errors reported by Bash).
 [_] Shellcheck did not report any errors.

Do NOT file a bug if you haven't ticked at least one box.

If you ticked both boxes, consider refining your sample code until only one
box is ticked, or explain why it's an actual problem."

**Comment by koalaman** [comment]
Meh, it's rarely an issue.

Parsing of regular and escaped line feeds is now improved. There's still a valid warning for this snippet, but removing the `\` before the `do` now parses correctly. Thanks!
